File: src/main/java/com/github/tanokun/tanorpg/player/quest/action/QuestSoundAction.java
package com.github.tanokun.tanorpg.player.quest.action;

import com.github.tanokun.tanorpg.TanoRPG;
import com.github.tanokun.tanorpg.player.Member;
import com.github.tanokun.tanorpg.util.command.CommandContext;
import org.bukkit.Bukkit;
import org.bukkit.Sound;

public class QuestSoundAction implements Action {
    private Sound sound;

    private int volume;
    private double pitch;

    public QuestSoundAction(CommandContext cc){
        sound = Sound.valueOf(cc.getArg(0, ""));
        volume = Integer.valueOf(cc.getArg(1, "1"));
        pitch = Double.valueOf(cc.getArg(2, "1"));
    }

    @Override
    public void execute(Member m) {
        TanoRPG.playSound(Bukkit.getPlayer(m.getUuid()), sound, volume, pitch);
    }
}

File: src/main/java/com/github/tanokun/tanorpg/player/quests/actions/QuestSoundAction.kt
package com.github.tanokun.tanorpg.player.quests.actions

import com.github.tanokun.tanorpg.TanoRPG
import com.github.tanokun.tanorpg.player.Member
import com.github.tanokun.tanorpg.player.quests.utils.annotation.ActionData
import com.github.tanokun.tanorpg.util.regex.RegexMatcher
import org.bukkit.Sound

@ActionData("sound", "<Sound:sound> <Int:volume> <Double:pitch>", "sound [Sound] [Volume] [Pitch]", "playerに音を流します")
class QuestSoundAction(regexMatcher: RegexMatcher, parameters: String): QuestAction {
    var sound: Sound = Sound.AMBIENT_CAVE
    var volume: Int = 1
    var pitch: Double = 1.0

    init {
        regexMatcher.matchResult(parameters) {
            sound = it.get("sound", Sound.AMBIENT_CAVE)
            volume = it.get("valume", 1)
            pitch = it.get("pitch", 1.0)
        }
    }

    override fun run(member: Member) = TanoRPG.playSound(member.player, sound, volume, pitch)
}